### PR TITLE
Fix error for old competitions

### PR DESCRIPTION
**Pull request by DanielEgdal**
Fixes an error from an old competition so it can now be marked as cancelled.

### CODE REVIEW

**Comment by danieljames-dj** [comment]
The changes looks good to me.

File: WcaOnRails/app/models/competition.rb
Comment (jonatanklosko) [inline WcaOnRails/app/models/competition.rb:384]: This only shows a warning box on the competition page in general, but doesn't block editing. I believe you intended to modify this part:

https://github.com/thewca/worldcubeassociation.org/blob/e974e9020e8c8a1e562c57695b96b312efb8eafa/WcaOnRails/app/models/competition.rb#L242-L249

Comment (DanielEgdal) [reply WcaOnRails/app/models/competition.rb]: You're right. I undid the first change (as it seems to only display before the competition is confirmed) and set the modification to the lines you mentioned.

File: WcaOnRails/app/models/competition.rb
Comment (gregorbg) [inline WcaOnRails/app/models/competition.rb:245]: Please add a comment that explains why you chose a specifc date in the middle of 2018 when this additional code gets added in 2021.

If there is no short'n'sweet explanation, at least add a comment along the lines of "refer to mail thread XY" or "based on a decision by WCAT" or something.

**Review by gregorbg — COMMENTED** [review]
Looks fine, but there's one point that I would like to discuss: Why do we perform this _exact same_ check in two places anyways?

It seems that Daniel was mistaken in the first place because it's not easy to tell the two apart. They even use the identical i18n key. Why not populate the `warnings[:schedule]` list based on the `errors` list that is created here?

**Comment by jonatanklosko** [comment]
> Why not populate the warnings[:schedule] list based on the errors list that is created here?

The warnings are global messages displayed to competition managers on all subpages of the given competitions and it's perfectly fine to show them for old competitions. The validations however result in form errors and prevent from completing the edit operation.

Sidenote: this specific validation should only run when confirming the competition (`if: :confirmed_at_changed?`). @DanielEgdal what was the exact path that triggered this error?

File: WcaOnRails/app/models/competition.rb
Comment (DanielEgdal) [reply WcaOnRails/app/models/competition.rb]: I changed this now to be 2018-12-31 as this is where all competitions required a schedule and added a reference why this is now needed per the email threads.

**Comment by DanielEgdal** [comment]
> Sidenote: this specific validation should only run when confirming the competition (`if: :confirmed_at_changed?`). @DanielEgdal what was the exact path that triggered this error?

Go this [page](https://www.worldcubeassociation.org/competitions/TokyoPyraminxOpen2018/edit/admin), and check the two boxes at the top ("Do not let organizers edit this competition" and "Make competition visible to the public").

**Comment by jonatanklosko** [comment]
Ah, that's because in this case we actually are confirming the competition, and it has no schedule, I see.

**Review by jonatanklosko — APPROVED** [review]

**Comment by DanielEgdal** [comment]
Is it ready to be merged then? (I don't have access to do merges after approvals)

**Comment by jonatanklosko** [comment]
@gregorbg feel free to merge unless you have any concerns :)

**Comment by gregorbg** [comment]
Done! :ship:

**Pull request merged**
